tests/test_ci_insights.py

```diff
@@ -179,18 +179,19 @@ def test_corge():
     # 1 span for the session and one per test.
     assert len(spans) == 1 + sum(outcomes.values())
 
-    for test_name, expected in {
-        "test_flaky_detection.py::test_foo": False,
-        "test_flaky_detection.py::test_bar": True,
-        "test_flaky_detection.py::test_baz": True,
-        "test_flaky_detection.py::test_qux": False,
-    }.items():
-        span = spans.get(test_name)
+    for span in spans.values():
+        for test, is_new_test in {
+            "test_flaky_detection.py::test_foo": False,
+            "test_flaky_detection.py::test_bar": True,
+            "test_flaky_detection.py::test_baz": True,
+            "test_flaky_detection.py::test_qux": False,
+        }.items():
+            if not span.name == test:
```

Comment:
I don't understand the loop, we are not just doing something like:

```suggestion
        map ={
            "test_flaky_detection.py::test_foo": False,
            "test_flaky_detection.py::test_bar": True,
            "test_flaky_detection.py::test_baz": True,
            "test_flaky_detection.py::test_qux": False,
        }
        
         if not map.get(span.name)
```

Reply:
My brain is burning 😶‍🌫️. Thanks, I simplified the test